streaming/src/java/org/apache/hive/streaming/HiveStreamingConnection.java

```diff
@@ -217,6 +216,11 @@ public static Builder newBuilder() {
     private int transactionBatchSize = DEFAULT_TRANSACTION_BATCH_SIZE;
     private boolean streamingOptimizations = DEFAULT_STREAMING_OPTIMIZATIONS_ENABLED;
     private RecordWriter recordWriter;
+    private Long writeId;
```

Comment:
nit: Builder seems to accept primitive types. May be use long and int primitives for writeId and statementId as well?